### PR TITLE
Fix #7457: DatePicker: lazy load disabled dates when panel is opened

### LINKED ISSUE #7457: DatePicker: lazy load disabled dates when panel is opened
**Describe the defect**
Lazy loading disabled dates was implemented in ticket #6355.
However, this doesn't work when opening the panel, no viewChangeEvent is fired. When changing months, the viewChangedEvent is fired correctly.

At the moment, when closing the panel, a viewChangeEvent is fired to set the the state of the dates of the current month to prepare the state the next time the panel is shown. However, this is not lazy. it should be initialized when showing the panel, not when hiding.

I think the following things should be changed;
When a LazyDateMetadataModel is used, ...
* a viewChangedEvent should be fired when the panel is opened.
* no viewChangedEvent should be fired when the panel is closed.

**Environment:**
 -  PF Version: 10.0
 -  JSF + version: Mojarra 2.3.9.SP09
 - Affected browsers: ALL (I could reproduce this with Chrome, Firefox and Edge, all updated to latest version. Haven't tested it with other browsers)

### PR DESCRIPTION
**Pull request by jepsar**
Fix #7457

@pmensalt please review

### CODE REVIEW

**Review by melloware — APPROVED** [review]

**Comment by christophs78** [comment]
@wal-jan provided a good amount of integrationtests with https://github.com/primefaces/primefaces/pull/7709. May we reuse some/most of them?

File: primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js [hunk 1]
```diff
@@ -107,11 +107,16 @@ PrimeFaces.widget.DatePicker = PrimeFaces.widget.BaseWidget.extend({
...
 
-                //user callback
+                // user callback
                 var preShow = $this.cfg.preShow;
                 if(preShow) {
                     return $this.cfg.preShow.call($this, inst);
                 }
+
+                // #7457 trigger view change if lazy model is used
+                if ($this.cfg.lazyModel) {
```
Comment (jepsar) [inline primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js:117]: I'm being conservative here. I think I'll drop this check and make the change global (consistent) and document it in the change log.

File: primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js [hunk 2]
```diff
@@ -308,6 +313,10 @@ PrimeFaces.widget.DatePicker = PrimeFaces.widget.BaseWidget.extend({
         var $this = this;
         this.cfg.onViewDateChange = function(event, date) {
             $this.viewDateOption = date;
+            // #7457 don't fire a change event when the picker is closed if lazy model is used
+            if ($this.cfg.lazyModel && event === null) {
```
Comment (jepsar) [inline primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js]: Same here

**Comment by melloware** [comment]
Yes it would be nice to add those integration tests!

**Comment by jepsar** [comment]
Agreed. I'll see if I can set it up. Hasn't been smooth sailing in the past and my laptop was upgraded (hardware) since...

**Comment by melloware** [comment]
I can add them @jepsar I will merge this PR for now.

**Comment by melloware** [comment]
Let me know when this is safe to merge.

File: primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js (hunk 1 above)
Comment (jepsar) [reply primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js:117]: @melloware what do you think? I'd like to remove these checks before merging the PR.

Comment (melloware) [reply primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js:117]: Hmmm so always fire the event?  Let's default to conservative for right now.

Comment (jepsar) [reply primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js:117]: Alright. Although I do think that firing a change on open and not in close is better in general.

Let me change `event === null` to `!event` and I think we are good to go.

Comment (jepsar) [reply primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js:117]: Done

**Comment by melloware** [comment]
OK i will add the integration tests for this. Thanks for the PR!

**Comment by wal-jan** [comment]
@christophs78 feel free to reuse any code I provided

**Pull request merged**